### ISSUE TITLE
Module fails to run with strict_variables set to true on Darwin for version 9.0.0

### ISSUE THREAD

**Issue by geoffdavis** [saz/puppet-ssh#334]
I get the following error when I run with `strict_variables`:

```none
Evaluation Error: Unknown variable: 'ssh::params::sshd_binary'. (file: /builds/anf-systems/puppet/puppet-environments/spec/fixtures/modules/ssh/manifests/init.pp, line: 165, column: 64)
```

This is presumably because `manifests/params.pp` does not have a variable definition for `sshd_binary` in the Darwin clause, nor for `sshd_environments_file`. See https://github.com/saz/puppet-ssh/blob/45b81357a036c087fbafdbe92a2ee8516c0aec26/manifests/params.pp#L62

Presumably this issue is gone in the master branch (I see that `params.pp` has disappeared), but the most recent forge module, 9.0.0, has this issue.

**Comment by geoffdavis** [comment]
Note that this can be worked around by setting several variables in your hieradata to define the missing parameters.

```yaml
---
# Darwin.yaml

# Work around bug in ssh 9.0.0 module
# See https://github.com/saz/puppet-ssh/issues/334
ssh::sshd_binary: /usr/sbin/sshd
ssh::sshd_environments_file: NULL
```

**Comment by saz** [comment]
@geoffdavis What's the reason for `ssh::sshd_environments_file: /dev/null`? It defaults to `undef` and wouldn't be added to the config file.

**Comment by geoffdavis** [comment]
It's been a while so I can't remember exactly why it was necessary. However, without it, some value was injected into the template by default.

It was either the undef assignment allowing a fall through to a default at a higher level of the hiera values, or the template evaluating undef as some garbage string value. I honestly don't remember which.

**Comment by geoffdavis** [comment]
I think ssh_environments_file had a conditional clause somewhere in the puppet code that broke if it wasn't set to something. It's been a while so I'll have to dig through the code again to find where. It wasn't the template. The /dev/null value was chosen because it was harmless.

**Comment by saz** [comment]
Thanks for the feedback, I'll set up a test installation and see what's needed to fix this completely

**Comment by saz** [comment]
This should be fixed in v10.0.0
